### DISCUSSION TITLE
Récupérer une couleur de la librairie dans un script

### DISCUSSION THREAD

**Discussion by JeremieDENAEGEL** [witsa/synapps#767] [Q&A] [answered]
Bonjour,

Est-il possible dans un OnValueChanged/OnPropertyChanged d'aller récupérer directement une couleur de la librairie sans passer par un additionnel ?

**Comment by Raphwit** [comment, marked as answer]
Bonjour, oui tout à fait les librairies son accessibles sur l'objet synapp, cet objet est posé sur les acteurs et les scènes.

Voici l'exemple d'une scène qui récupère la variable "cyan" dans la librairie de couleur et qui l'assigne dans un acteur texte, on note que le contenu de l'acteur texte n'est pas défini grâce à une additionnelle !

```
SYNAPPS-STUDIO-SCENE|{"config":{"key":"scene2","name":"Scène 2"},"leadActor":{"type":"layout/stack","key":"stack1","children":[{"type":"display/text","key":"text1","properties":{"content":"Text"},"events":{"onPostInit":["this.properties.content = this.synapp.colors.cyan"]}}]}}
```
